subsys/bootloader/cmake/sign.cmake

```diff
@@ -77,23 +77,36 @@ foreach (slot ${slots})
   set(signed_hex ${PROJECT_BINARY_DIR}/signed_by_b0_${slot}.hex)
   set(signed_bin ${PROJECT_BINARY_DIR}/signed_by_b0_${slot}.bin)
 
-  set(sign_depends ${PROJECT_BINARY_DIR}/${slot}.hex ${SIGN_KEY_FILE_DEPENDS})
+  set(sign_depends ${SIGN_KEY_FILE_DEPENDS})
 
   if (CONFIG_BUILD_S1_VARIANT AND NOT CONFIG_BOOTLOADER_MCUBOOT AND
       CONFIG_SPM AND ("${slot}" STREQUAL s1_image))
+    set(to_sign_incorrect_size ${PROJECT_BINARY_DIR}/s1_spm_app.hex)
     list(APPEND sign_depends s1_spm_app_hex_target)
   else()
+    set(to_sign_incorrect_size ${PROJECT_BINARY_DIR}/${slot}.hex)
     if(DEFINED ${slot}_is_from_child_image)
       list(APPEND sign_depends ${${slot}_is_from_child_image}_subimage)
     else()
       list(APPEND sign_depends ${slot}_hex)
     endif()
   endif()
 
-  set(to_sign ${PROJECT_BINARY_DIR}/${slot}.hex)
+  list(APPEND sign_depends ${to_sign_incorrect_size})
+  set(to_sign ${PROJECT_BINARY_DIR}/${slot}_fw_info_updated.hex)
   set(hash_file ${GENERATED_PATH}/${slot}_firmware.sha256)
   set(signature_file ${GENERATED_PATH}/${slot}_firmware.signature)
 
+  set(size_fix_cmd
+    ${PYTHON_EXECUTABLE}
+    ${NRF_BOOTLOADER_SCRIPTS}/fix_size.py
+    --magic-value "${FIRMWARE_INFO_MAGIC}"
```

Comment:
This will be empty if `CONFIG_FW_INFO=n`, is that acceptable ?

Reply:
This is only included if `SECURE_BOOT` is enabled, and SECURE_BOOT selects FW_INFO so I guess there is a dependency there which makes FW_INFO always `y` if this is executed?

Reply:
ok, a bit hard to know based on just this PR, thanks for clarifying.